user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+  private final MessageSourceNotificationFactory notificationFactory;
+
+  private enum AcceptedFileType {
+    EXCEL("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"),
+    TSV("text/tab-separated-values", "text/plain");
+    private final List<String> mimeTypes;
+
+    AcceptedFileType(String... mimeTypes) {
+      this.mimeTypes = List.of(mimeTypes);
+    }
+
+    static Optional<AcceptedFileType> forMimeType(String mimeType) {
+      return Arrays.stream(values())
+          .filter(it -> it.mimeTypes.contains(mimeType))
+          .findFirst();
+    }
+
+    static Set<String> allMimeTypes() {
+      return Arrays.stream(values()).flatMap(it -> it.mimeTypes.stream())
+          .collect(Collectors.toUnmodifiableSet());
+    }
+  }
+
+  private final EditableMultiFileMemoryBuffer uploadBuffer;
+  private final List<MeasurementFileItem> measurementFileItems;
+
+  public MeasurementUpload(
+      AsyncProjectService service,
+      Context context,
+      MetadataConverterV2<? extends ValidationRequestBody> converter,
+      MessageSourceNotificationFactory notificationFactory) {
+    this.uploadBuffer = new EditableMultiFileMemoryBuffer();
+    this.measurementFileItems = new ArrayList<>();
+    this.service = requireNonNull(service);
+    this.context = requireNonNull(context);
+    this.validationProgress = new Div();
+    this.converter = requireNonNull(converter);
+    this.notificationFactory = requireNonNull(notificationFactory);
+    var upload = new Upload(uploadBuffer);
+    upload.setAcceptedFileTypes(AcceptedFileType.allMimeTypes().toArray(String[]::new));
+    upload.setMaxFileSize(MAX_FILE_SIZE_BYTES);
+    upload.addSucceededListener(this::onUploadSucceeded);
+    upload.addFileRejectedListener(this::onFileRejected);
+    upload.addFailedListener(this::onUploadFailed);
+    upload.addFileRemovedListener(this::onFileRemoved);
+    this.uploadItemsDisplay = new UploadItemsDisplay(upload);
+    refreshFileItemDisplay();
+    add(uploadItemsDisplay);
+    add(validationProgress);
+  }
+
+  private void refreshFileItemDisplay() {
+    if (measurementFileItems.isEmpty()) {
+      uploadItemsDisplay.hide();
+      return;
+    }
+    uploadItemsDisplay.show();
+  }
+
+  private void removeFile(String fileName) {
+    uploadBuffer.remove(fileName);
+    measurementFileItems.removeIf(
+        measurementFileItem -> measurementFileItem.fileName().equals(fileName));
+    uploadItemsDisplay.removeFileFromDisplay(fileName);
+    refreshFileItemDisplay();
+  }
+
+  private void onFileRemoved(FileRemovedEvent fileRemovedEvent) {
+    removeFile(fileRemovedEvent.getFileName());
+  }
+
+  private void onUploadFailed(FailedEvent failedEvent) {
+    displayError();
+  }
+
+  private void displayError(String message) {
+    getUI().ifPresent(ui -> ui.access(() -> {
+      var toast = notificationFactory.toast("measurement.upload.failed.reason",
```

Comment:
I think we could list the reasons in the message property file instead. `measurement.upload.failed.unsupported-file-type` or similar. Otherwise we have the error message in the code again and do not gain from the message source here.